src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -14,14 +16,15 @@ object UserService {
     def pure[A](a: A): DBIO[A] = DBIO.successful(a)
     def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
   }
-
+  
   def createUser(createdUser: User)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], UserId]] = {
     val resT = for {
       validUser <- XorT[DBIO, NonEmptyList[ValidationError], User](User.validateUser(createdUser.email, createdUser.password))
-      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(validUser))
+      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(hashUserPassword(validUser)))
     } yield userId
 
     resT.value
   }
 
+  private def hashUserPassword(user: User): User = user.copy(password = user.password.bcrypt)
```

Comment:
`Should the "validateUser" be responsible for hashing the password` - good point - if we would go this road the `vaidateUser` would have to be named differently. I would see it more as a factory method then a validating method.

`f the password will be at all times in its hashed form in the DB then that should be reflected in the mapping class User` - yes, the field would probably be better named as `passwordHash` in the db.

`the password should be hashed by the User class whenever an instance is created` - kind of yes. But not necessarily by the User class but as I mentioned above - a factory method in a companion object.

`And we would have 3 classes to basically represent the same thing` - yes. I feel your doubts :). I do have them too. But if you think about it, it's not necessarily `representing the same thing`. We are representing different states in the most accurate way we can that will prevent the model from being in an inconsistent state. There will be more benefits I hope.